client/actions/usersActions.js

```diff
@@ -0,0 +1,540 @@
+ * @function userModifySuccessful
+ *
+ * @param {any} payload
+ * @return {object} Action
+ */
+const userModifySuccessful = payload => ({
+  type: actionTypes.USER_MODIFY_SUCCESSFUL,
+  payload
+});
+
+/**
+ * Dispatches a USER_MODIFY_FAILED action
+ * @function userModifyFailed
+ *
+ * @param {any} payload
+ * @return {object} Action
+ */
+const userModifyFailed = payload => ({
+  type: actionTypes.USER_MODIFY_FAILED,
+  payload
+});
+
+/**
+ * Dispatches a USER_DELETE_REQUEST action
+ * @function userDeleteRequest
+ *
+ * @return {object} Action
+ */
+const userDeleteRequest = () => ({
+  type: actionTypes.USER_DELETE_REQUEST
+});
+
+/**
+ * Dispatches a USER_DELETE_SUCCESSFUL action
+ * @function userDeleteSuccessful
+ *
+ * @param {any} payload
+ * @return {object} Action
+ */
+const userDeleteSuccessful = payload => ({
+  type: actionTypes.USER_DELETE_SUCCESSFUL,
+  payload
+});
+
+/**
+ * Dispatches a USER_DELETE_FAILED action
+ * @function userDeleteFailed
+ *
+ * @param {any} payload
+ * @return {object} Action
+ */
+const userDeleteFailed = payload => ({
+  type: actionTypes.USER_DELETE_FAILED,
+  payload
+});
+
+/**
+ * Generate and return the error messages in HTML format
+ * @function getErrorMessage
+ *
+ * @param {any} errors
+ * @returns {string} errorMessage
+ */
+const getErrorMessage = (errors) => {
+  const errorMessage = lodash.reduce(errors, (result, error) =>
+    `${error}<br/>${result}`
+  , '');
+  return errorMessage;
+};
+
+const usersActions = {
+  setTokenToLocalStorage(user, token) {
+    window.localStorage.setItem('token', token);
+    window.localStorage.setItem('user', JSON.stringify({
+      id: user.id,
+      username: user.username
+    }));
+    return usersActions.addUser(user, token);
+  },
+
+  /**
+   * Dispatches actions associated with adding new user
+   * @function addUser
+   *
+   * @param {any} user
+   * @param {any} token
+   * @param {any} callback
+   * @returns {object} Action
+   */
+  addUser(user, token, callback) {
+    axios.defaults.headers.common.Authorization = `bearer ${token}`;
+    if (Object.keys(user).length < 3) {
+      axios.get(`/api/v1/users/${user.id}`)
+      .then(response => (
+        callback({
+          type: actionTypes.ADD_USER,
+          payload: response.data.user
+        })));
+    } else {
+      return ({
+        type: actionTypes.ADD_USER,
+        payload: user
+      });
+    }
+  },
+
+  /**
+   * Dispatches actions associated with signing user up
+   * @function signUserUp
+   *
+   * @param {any} userData
+   * @returns {object} Action
+   */
+  signUserUp({
+    firstname, lastname, username, email, password, roleId, confirmPassword
+  }) {
+    const userdata = {
+      firstname, lastname, username, email, password, roleId, confirmPassword
+    };
+    const validation = Validator.validateSignUp(userdata);
+
+    return (dispatch) => {
+      dispatch(requestSignup(userdata.username));
+      if (validation.isValid) {
```

Comment:
I think you should validate your data before sending to your actions. Gives you less bugs to think about in the actions